psi4/src/psi4/libfock/DFJCOSK.cc

```diff
@@ -239,26 +255,35 @@ void DFJCOSK::common_init() {
     };
     grid_final_ = std::make_shared<DFTGrid>(primary_->molecule(), primary_, grid_final_int_options, grid_final_str_options, grid_final_float_options, options_);
 
-    // Sanity-check of grids to ensure no negative grid weights
-    // COSX crashes when grids with negative weights are used,
+    // Print out warning if grid with negative grid weights is used
+    // Original Nesse COSX formulation does not support negative grid weights
     // which can happen with certain grid configurations
-    // See https://github.com/psi4/psi4/issues/2890
+    // the Psi4 COSX implementation is slightly modified to work with negative grid weights
+    // See https://github.com/psi4/psi4/issues/2890 for discussion
+    auto warning_printed_init = false;
     for (const auto &init_block : grid_init_->blocks()) {
         const auto w = init_block->w();
         for (int ipoint = 0; ipoint < init_block->npoints(); ++ipoint) {
             if (w[ipoint] < 0.0) {
-	      throw PSIEXCEPTION("The definition of the current initial grid includes negative weights. As these are not suitable for the COSX implementation, please choose another initial grid through adjusting either COSX_PRUNING_SCHEME or COSX_SPHERICAL_POINTS_INITIAL.");
+	        outfile->Printf("  WARNING: The definition of the current initial grid includes negative weights, which the original COSX formulation does not support!\n    If this is of concern, please choose another initial grid through adjusting either COSX_PRUNING_SCHEME or COSX_SPHERICAL_POINTS_INITIAL.\n\n");
```

Comment:
I think this should not be a WARNING but INFO (does this exist in Psi4?)

Reply:
I'm happy to switch to INFO, but Psi4 seems to trend towards using WARNING instead of INFO for printout like this, at least from my observation.

Reply:
Added! I'll leave the conversation open, though, in case others want to comment on INFO vs. WARNING.

Reply:
> I think this should not be a WARNING but INFO (does this exist in Psi4?)

There's no proper logging from c-side. :-( Only `PSIEXCEPTION` that stops execution and `Printf("WARNING/INFO/ETC` that goes to the output file. Admittedly not great. I don't think the situation has been seriously examined for improvement since pybind11 (and having python on the outside of psi4) was young.

> in case others want to comment on INFO vs. WARNING.

I agree it's probably INFO at this point (rounding the grid up to down to a pure-positive one would be WARNING), but there's not standardization.

Reply:
> > I think this should not be a WARNING but INFO (does this exist in Psi4?)
> 
> There's no proper logging from c-side. :-( Only `PSIEXCEPTION` that stops execution and `Printf("WARNING/INFO/ETC` that goes to the output file. Admittedly not great. I don't think the situation has been seriously examined for improvement since pybind11 (and having python on the outside of psi4) was young.
> 
> > in case others want to comment on INFO vs. WARNING.
> 
> I agree it's probably INFO at this point (rounding the grid up to down to a pure-positive one would be WARNING), but there's not standardization.

All good to know! Maybe such logging is something that could be more seriously looked at in the future. It wouldn't be a bad functionality to have.